spring-boot-autoconfigure/src/test/java/org/springframework/boot/autoconfigure/security/oauth2/client/OAuth2ClientAutoConfigurationTests.java

```diff
@@ -0,0 +1,630 @@
+import org.springframework.security.oauth2.client.authentication.AuthorizationCodeAuthenticationProvider;
+import org.springframework.security.oauth2.client.authentication.AuthorizationCodeRequestRedirectFilter;
+import org.springframework.security.oauth2.client.registration.ClientRegistration;
+import org.springframework.security.oauth2.client.registration.ClientRegistrationRepository;
+import org.springframework.security.oauth2.client.user.nimbus.NimbusOAuth2UserService;
+import org.springframework.security.oauth2.core.AuthorizationGrantType;
+import org.springframework.security.oauth2.core.ClientAuthenticationMethod;
+import org.springframework.security.web.DefaultSecurityFilterChain;
+import org.springframework.security.web.FilterChainProxy;
+import org.springframework.security.web.util.matcher.AnyRequestMatcher;
+import org.springframework.util.ReflectionUtils;
+import org.springframework.web.context.support.AnnotationConfigWebApplicationContext;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+/**
+ * Tests for {@link OAuth2ClientAutoConfiguration}.
+ *
+ * @author Joe Grandja
+ */
+public class OAuth2ClientAutoConfigurationTests {
+	private static final String CLIENT_PROPERTY_PREFIX = "security.oauth2.client";
+
+	private static final String CLIENT_ID_PROPERTY = "client-id";
+
+	private static final String CLIENT_SECRET_PROPERTY = "client-secret";
+
+	private static final String DEFAULT_REDIRECT_URI = "{scheme}://{serverName}:{serverPort}{contextPath}/oauth2/authorize/code/{clientAlias}";
+
+	private static final String GOOGLE_CLIENT_KEY = "google";
+
+	private static final String GOOGLE_CLIENT_ALIAS = GOOGLE_CLIENT_KEY;
+
+	private static final String GOOGLE_CLIENT_PROPERTY_BASE = CLIENT_PROPERTY_PREFIX + "."
+			+ GOOGLE_CLIENT_KEY;
+
+	private static final String GITHUB_CLIENT_KEY = "github";
+
+	private static final String GITHUB_CLIENT_ALIAS = GITHUB_CLIENT_KEY;
+
+	private static final String GITHUB_CLIENT_PROPERTY_BASE = CLIENT_PROPERTY_PREFIX + "."
+			+ GITHUB_CLIENT_KEY;
+
+	private static final String OKTA_CLIENT_KEY = "okta";
+
+	private static final String OKTA_CLIENT_ALIAS = OKTA_CLIENT_KEY;
+
+	private static final String OKTA_CLIENT_PROPERTY_BASE = CLIENT_PROPERTY_PREFIX + "."
+			+ OKTA_CLIENT_KEY;
+
+	private static final String FACEBOOK_CLIENT_KEY = "facebook";
+
+	private static final String FACEBOOK_CLIENT_ALIAS = FACEBOOK_CLIENT_KEY;
+
+	private static final String FACEBOOK_CLIENT_PROPERTY_BASE = CLIENT_PROPERTY_PREFIX
+			+ "." + FACEBOOK_CLIENT_KEY;
+
+	private final SpringApplication application = new SpringApplication();
+
+	private final ConfigFileApplicationListener environmentPostProcessor = new ConfigFileApplicationListener();
+
+	private AnnotationConfigWebApplicationContext context;
+
+	@After
+	public void close() {
+		if (this.context != null) {
+			this.context.close();
+		}
+	}
+
+	@Test
+	public void refreshContextWhenNoOverridesThenLoadDefaultConfiguration()
```

Comment:
If it makes sense to do so, it'd be nice to split this test up a bit

Reply:
The tests will be much smaller now given that I split up the auto-config's into 2 classes.